src/Microsoft.TestPlatform.ObjectModel/Adapter/Interfaces/ITestExecutor2.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel.Adapter
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Defines the test executor which provides capability to run tests.  
+    /// 
+    /// A class that implements this interface will be available for use if its containing 
+    //  assembly is either placed in the Extensions folder or is marked as a 'UnitTestExtension' type 
+    //  in the vsix package.
+    /// </summary>
+    public interface ITestExecutor2 : ITestExecutor
+    {
+        /// <summary>
+        /// Runs only the tests specified by parameter 'tests'. 
+        /// </summary>
+        /// <param name="tests">Tests to be run.</param>
+        /// <param name="runContext">Context to use when executing the tests.</param>
+        /// <param param name="frameworkHandle">Handle to the framework to record results and to do framework operations.</param>
+        void RunTests(IEnumerable<TestCase> tests, IRunContext runContext, IFrameworkHandle2 frameworkHandle);
+
+        /// <summary>
+        /// Runs 'all' the tests present in the specified 'sources'. 
+        /// </summary>
+        /// <param name="sources">Path to test container files to look for tests in.</param>
+        /// <param name="runContext">Context to use when executing the tests.</param>
+        /// <param param name="frameworkHandle">Handle to the framework to record results and to do framework operations.</param>
+        void RunTests(IEnumerable<string> sources, IRunContext runContext, IFrameworkHandle2 frameworkHandle);
```

Comment:
Agreed with both points. If we remove the overloads we hurt discoverability, otherwise there'll be 4 overloads to implement, and per our discussion, if sometime in the future we decide an `IFrameworkHandle3` is needed, we're going to have a really bloated `ITestExecutor` interface.

If we remove the overloads there'll be no casting involved on the test platform side though, because we can pass the `FrameworkHandle` which implements the latest iteration of `IFrameworkHandle` interface, and neither will be on the adapter side, because of the same reason. Old adapters will be oblivious to changes on the `FrameworkHandle` (because most certainly they won't need them anyways), while new adapters don't need to cast the handle in order to use latest features. So how about we remove the overloads altogether while leaving `ShouldAttachToTestHost` methods there, @shyamnamboodiripad ?

Reply:
As per our discussion @shyamnamboodiripad and taking into account @AbhitejJohn's input, I removed the overloads. Pending TAEF team input they might get resurrected.